python-backend/app/api/mines/reports/resources/mine_report_comment.py

```diff
@@ -0,0 +1,132 @@
+            raise NotFound('No mine report submissions for this report')
+
+        current_app.logger.info(f'Retrieving comments for {mine_report}')
+
+        comments = [comment.__dict__ for submission in mine_report_submissions[:-1] for comment in submission.comments]
+        for comment in comments:
+            comment['from_latest_submission'] = False
+        latest_comments = [comment.__dict__ for submission in mine_report_submissions[-1:] for comment in submission.comments]
+        for comment in latest_comments:
+            comment['from_latest_submission'] = True
+
+        return comments + latest_comments, 200
+
+    @api.expect(MINE_REPORT_COMMENT_MODEL)
+    @api.doc(description='creates a new comment for the report submission')
+    @api.marshal_with(MINE_REPORT_COMMENT_MODEL, code=201)
+    @requires_role_edit_report
+    def post(self, mine_guid, mine_report_guid):
+
+        mine_report_submission = MineReportSubmission.find_latest_by_mine_report_guid(
+            mine_report_guid)
+
+        if not mine_report_submission:
+            raise NotFound('Mine report submission not found')
+            # TODO: Do we want to create a submission if it doesn't exist?
+
+        data = self.parser.parse_args()
+
+        if not data['report_comment']:
+            raise BadRequest('Empty comment')
+
+        mine_report_comment_guid = uuid.uuid4()
+
+        mine_report_comment = MineReportComment.create(
+            mine_report_submission,
+            mine_report_comment_guid=mine_report_comment_guid,
+            report_comment=data['report_comment'],
+            comment_visibility_ind=data['comment_visibility_ind'],
+        )
+
+        current_app.logger.info(f'Creating comment {mine_report_comment}')
+
+        try:
+            mine_report_comment.save()
+        except Exception as e:
+            raise InternalServerError(f'Error when saving: {e}')
+
+        return mine_report_comment, 201
+
+
+class MineReportCommentResource(Resource, UserMixin):
+    parser = CustomReqparser()
+    parser.add_argument('report_comment', type=str, location='json')
+    parser.add_argument('comment_visibility_ind', type=inputs.boolean, location='json')
+
+    @api.expect(MINE_REPORT_COMMENT_MODEL)
+    @api.doc(description='update a comment')
+    @api.marshal_with(MINE_REPORT_COMMENT_MODEL, code=201)
+    @requires_role_edit_report
+    def put(self, mine_guid, mine_report_guid, mine_report_comment_guid=None):
+
+        data = self.parser.parse_args()
+        comment = MineReportComment.find_by_guid(mine_report_comment_guid)
+        if not comment:
+            raise NotFound('Mine Report Comment not found.')
+
+        current_app.logger.info(f'Updating {comment} with {data}')
+        for key, value in data.items():
+            setattr(comment, key, value)
+
+        try:
+            comment.save()
+        except Exception as e:
+            raise InternalServerError(f'Error when saving: {e}')
```

Comment:
Is this save particularly problematic or something? I'm curious why it has a try/except wrapper when basically no other save invocations in our app do

Reply:
I think its just idiomatic for me: wrapping any persistence layer or remote calls in a try block. Looking at it now it doesn't look consistent everywhere in our codebase. I'll remove for now and look to see if there's any reason we'd want to do it this way.